code/modules/shuttle/supply.dm

```diff
@@ -746,6 +776,58 @@
 
 	frequency.post_signal(src, status_signal)
 
+/obj/machinery/computer/supplycomp/proc/notify_pda(recipient, order_number, ordered_object, multiple = FALSE, confirmed = FALSE)
+
+	var/obj/machinery/message_server/useMS = null
+	if(GLOB.message_servers)
+		for(var/A in GLOB.message_servers)
+			var/obj/machinery/message_server/MS = A
+			if(MS.active)
+				useMS = MS
+				break
+
+	if(!useMS)
+		return
+
+	var/datum/data/pda/app/messenger/PM
+	var/obj/item/pda/P
+	for(var/A in GLOB.PDAs)
+		P = A
+		if(P.owner == recipient)
+			PM = P.find_program(/datum/data/pda/app/messenger)
+			break
+
+	if(!PM || PM.toff)
+		return
+
+	var/turf/sender_pos = get_turf(src)
+	var/turf/recipient_pos = get_turf(P)
+
+	var/sendable = FALSE
+	var/receivable = FALSE
+	for(var/obj/machinery/tcomms/core/C in GLOB.tcomms_machines)
+		if(C.zlevel_reachable(sender_pos.z))
+			sendable = TRUE
+		if(C.zlevel_reachable(recipient_pos.z))
+			receivable = TRUE
+		// Once both are done, exit the loop
+		if(sendable && receivable)
+			break
+
+	if(!sendable || !receivable)
+		return	
+
+	var/havehas = "has"
+	var/s
+	var/confirmverb = "rejected"
+	if(confirmed)
+		confirmverb = "approved"
+	if(multiple)
+		havehas = "have"
+		s = "s"
+
+	useMS.send_pda_message("[recipient]","Supply Ordering Console","Your order[s] #[order_number] ([ordered_object]) [havehas] been [confirmverb]")
+	PM.notify("<b>Automatic message from Supply Ordering Console, </b>\"Your order[s] #[order_number] ([ordered_object]) [havehas] been [confirmverb]\" (No reply)", 0)
```

Comment:
Correct me if I'm wrong but, since you are doing a for loop in the `"rejectall"` section under line 693, each of these orders are printed one by one. Which means you have no use for the `multiple` variable here, as it will always be a singular order.

Reply:
In the second screenshot there is a message which contains multiple orders. What happens is `orders_by_users` is an associative list, where the index is "requester name+requested object" and the value is another list which contains `var/datum/supply_order/` datums with the same object and requestd by the same person, which are then put into the same message.
The code from line 707 to 715 fills the message.

Reply:
Yeah sorry I see what you did there now.